### PR TITLE
Add pronuncation to RouteStep

### PR DESCRIPTION
**Pull request by TheMarex**

### CODE REVIEW

**Comment by emiltin** [comment]
this is a breaking change, since it requires front-ends to parse the name string instead of simply displaying it?

**Comment by emiltin** [comment]
if possible i think it would be better to implement this via #1342.

we're already have to pack things like the highway tag into the name (using "My Way {'highway':'primary'}" for cases where the name has no name. now we're also packing name:pronounciation and would end up with "MyWay#meyeway{'highway':'primary'}".

File: taginfo.json [hunk 1]
```diff
@@ -57,6 +57,7 @@
         {"key": "maxspeed:advisory"},
         {"key": "maxspeed:advisory:forward"},
         {"key": "maxspeed:advisory:backward"},
+        {"key": "name:pronounciation", "description": "Pronounciation hit for names", "object_types": ["way"]},
```
Comment (daniel-j-h) [inline taginfo.json]: _hint_

**Comment by systemed** [comment]
Should presumably be `name:pronunciation` (not `name:pronounciation`).

**Comment by TheMarex** [comment]
@systemed damn my bad spelling. Should be fixed now.

@emiltin I found a way to do this without the hack and make it compatible with all clients. There is now a new property `pronunciation`.

File: include/extractor/extraction_way.hpp [hunk 2]
```diff
@@ -46,6 +46,7 @@ struct ExtractionWay
     double backward_speed;
     double duration;
     std::string name;
+    std::string pronunciation;
```
Comment (daniel-j-h) [inline include/extractor/extraction_way.hpp]: You also have to adapt the `.clear()` function for this new member attribute.

File: include/engine/datafacade/internal_datafacade.hpp [hunk 3]
```diff
@@ -587,6 +587,11 @@ class InternalDataFacade final : public BaseDataFacade
         return result;
     }
 
+    std::string GetPronunciationForID(const unsigned name_id) const override final
+    {
+        return GetNameForID(name_id + 1);
```
Comment (daniel-j-h) [inline include/engine/datafacade/internal_datafacade.hpp]: Doing the subsequent copies into the char data and using the name_id as base here is _really_ clever --- we should document that somewhere!

File: include/extractor/extraction_way.hpp (hunk 2 above)
Comment (TheMarex) [inline include/extractor/extraction_way.hpp]: Good catch!

**Pull request merged**